app/src/main/contrib/components/Responsibilities/Overview/ConfidentialityAgreement.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from "react";
+import {ChangeEvent} from "react";
+import {settings} from "../../../../shared/Settings";
+import {connect} from "react-redux";
+import {branch, compose, renderComponent} from "recompose";
+import {Dispatch} from "redux";
+import {ContribAppState} from "../../../reducers/contribAppReducers";
+import {default as withLifecycle, LifecycleMethods} from "@hocs/with-lifecycle";
+import {userActionCreators} from "../../../actions/userActionCreators";
+import {LoadingElement} from "../../../../shared/partials/LoadingElement/LoadingElement";
+
+interface State {
+    checked: boolean;
+}
+
+const fullConfidentialityAgreement = require('./rfp-applicants-confidentiality.pdf');
+
+export class ConfidentialityAgreementComponent extends React.Component<ConfidentialityProps, State> {
+
+    constructor() {
+        super();
+        this.state = {checked: false};
+        this.onChange = this.onChange.bind(this)
+    }
+
+    onChange(e: ChangeEvent<HTMLInputElement>) {
+
+        this.setState({
+            checked: e.target.checked
+        })
+    }
+
+    render() {
+
+        if (this.props.signed == null) {
```

Comment:
Have moved half of the branching logic out of recompose and into the component's render method. Hopefully this is easier to follow now!